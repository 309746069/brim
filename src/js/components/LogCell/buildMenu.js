/* @flow */

import * as searchBar from "../../actions/searchBar"
import * as packets from "../../actions/packets"
import * as logDetails from "../../actions/logDetails"
import * as timeWindow from "../../actions/timeWindow"
import type {Props} from "./LogCellActions"
import {TimeField} from "../../models/Field"

type Action = {
  type: "action",
  text: string,
  onClick: Event => void
}

type Seperator = {
  type: "seperator"
}

export type MenuItemData = Seperator | Action

export default ({dispatch, field, space, log}: Props): MenuItemData[] => {
  if (log.isPath("conn") && space.packet_support) {
    return [
      exclude(field, dispatch),
      include(field, dispatch),
      countBy(field, dispatch),
      seperator(),
      detail(log, dispatch),
      pcaps(log, dispatch)
    ]
  }

  if (field instanceof TimeField) {
    return [
      fromTime(field, dispatch),
      toTime(field, dispatch),
      seperator(),
      detail(log, dispatch)
    ]
  }

  return [
    exclude(field, dispatch),
    include(field, dispatch),
    countBy(field, dispatch),
    seperator(),
    detail(log, dispatch)
  ]
}

const exclude = (field, dispatch) => ({
  type: "action",
  text: "Filter out these values",
  onClick: (e: Event) => {
    e.stopPropagation()
    dispatch(searchBar.appendQueryExclude(field))
    dispatch(searchBar.submitSearchBar())
  }
})

const include = (field, dispatch) => ({
  type: "action",
  text: "Only show these values",
  onClick: (e: Event) => {
    e.stopPropagation()
    dispatch(searchBar.appendQueryInclude(field))
    dispatch(searchBar.submitSearchBar())
  }
})

const countBy = (field, dispatch) => ({
  type: "action",
  text: `Count by ${field.name}`,
  onClick: (e: Event) => {
    e.stopPropagation()
    dispatch(searchBar.appendQueryCountBy(field))
    dispatch(searchBar.submitSearchBar())
  }
})

const pcaps = (log, dispatch) => ({
  type: "action",
  text: "Download PCAPS",
  onClick: (_e: Event) => {
    dispatch(packets.fetchPackets(log))
  }
})

const detail = (log, dispatch) => ({
  type: "action",
  text: "View log details",
  onClick: () => {
    dispatch(logDetails.viewLogDetail(log))
  }
})

const fromTime = (field, dispatch) => ({
  type: "action",
  text: "Use as \"start\" time",
  onClick: () => {
    dispatch(timeWindow.setOuterFromTime(field.toDate()))
    dispatch(searchBar.submitSearchBar())
  }
})

const toTime = (field, dispatch) => ({
  type: "action",
  text: "Use as \"end\" time",
  onClick: () => {
    dispatch(timeWindow.setOuterToTime(field.toDate()))
    dispatch(searchBar.submitSearchBar())
  }
})

const seperator = () => ({
  type: "seperator"
})
