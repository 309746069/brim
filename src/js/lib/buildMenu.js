/* @flow */

import type {Space} from "../lib/Space"
import Log from "../models/Log"
import Field from "../models/Field"
import {type ResultsTabEnum} from "../reducers/view"
import buildLogRightClickMenu from "./buildLogRightClickMenu"
import buildAnalyticRightClickMenu from "./buildAnalyticRightClickMenu"

type Args = {
  dispatch: Function,
  field: Field,
  log: Log,
  space: Space,
  resultType: ResultsTabEnum
}

export default (args: Args) => {
  if (args.resultType === "logs") {
    return buildLogRightClickMenu(args)
  } else {
    return buildAnalyticRightClickMenu(args)
  }
}
