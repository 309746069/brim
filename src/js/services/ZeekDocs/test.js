/* @flow */
import ZeekDocs from "./"

test("describe conn uid", () => {
  let desc = ZeekDocs.describe("conn", "uid")

  expect(desc).toBe("A unique identifier of the connection.")
})
