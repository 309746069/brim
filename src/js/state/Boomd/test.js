/* @flow */
import Boomd from "./"
import initTestStore from "../../test/initTestStore"

test("setting use analytics cache", () => {
  const store = initTestStore()
  store.dispatch(Boomd.useCache(true))
  expect(Boomd.usingCache(store.getState())).toEqual(true)
  store.dispatch(Boomd.useCache(false))
  expect(Boomd.usingCache(store.getState())).toEqual(false)
})

test("setting use index", () => {
  const store = initTestStore()
  store.dispatch(Boomd.useIndex(true))
  expect(Boomd.usingIndex(store.getState())).toEqual(true)
  store.dispatch(Boomd.useIndex(false))
  expect(Boomd.usingIndex(store.getState())).toEqual(false)
})
