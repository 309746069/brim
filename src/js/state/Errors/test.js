/* @flow */

import AppError from "../../models/AppError"
import Errors from "./"
import initTestStore from "../../test/initTestStore"

let store
beforeEach(() => {
  store = initTestStore()
})

test("ERROR_CREATE", () => {
  let state = store.dispatchAll([Errors.createError("Bug")])

  expect(Errors.getErrors(state)[0]).toEqual(expect.any(AppError))
})

test("ERRORS_CLEAR", () => {
  let state = store.dispatchAll([
    Errors.createError("Bug"),
    Errors.createError("Bug2"),
    Errors.clearErrors()
  ])

  expect(Errors.getErrors(state)).toEqual([])
})
