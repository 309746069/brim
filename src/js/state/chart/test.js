/* @flow */

import chart from "./"
import initTestStore from "../../test/initTestStore"

let store
beforeEach(() => {
  store = initTestStore()
})

let records = [
  [{name: "id", type: "string", value: "1"}],
  [{name: "id", type: "string", value: "2"}]
]

test("chart records append", () => {
  let state = store.dispatchAll([
    chart.appendRecords([records[0]]),
    chart.appendRecords([records[1]])
  ])

  expect(chart.getRecords(state)).toEqual(records)
})

test("chart records status", () => {
  let state = store.dispatchAll([chart.setStatus("SUCCESS")])

  expect(chart.getStatus(state)).toBe("SUCCESS")
})

test("chart records clear", () => {
  let state = store.dispatchAll([
    chart.appendRecords(records),
    chart.setStatus("SUCCESS"),
    chart.clear()
  ])

  expect(chart.getStatus(state)).toBe("INIT")
  expect(chart.getRecords(state)).toEqual([])
})
