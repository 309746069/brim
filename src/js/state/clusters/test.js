/* @flow */

import {addCluster, removeCluster, setCluster, setClusterError} from "./actions"
import {
  getCurrentCluster,
  getClusterMessage,
  getSavedClusters
} from "./selectors"
import initTestStore from "../../test/initTestStore"

let store
beforeEach(() => {
  store = initTestStore()
})

let cluster = {
  host: "boom.com",
  port: "9867",
  username: "kerr",
  password: "123"
}

test("addCluster", () => {
  let state = store.dispatchAll([addCluster(cluster)])

  expect(getSavedClusters(state)).toEqual([cluster])
})

test("addCluster when it already exists", () => {
  let state = store.dispatchAll([addCluster(cluster), addCluster(cluster)])

  expect(getSavedClusters(state)).toEqual([cluster])
})

test("removeCluster", () => {
  let state = store.dispatchAll([addCluster(cluster), removeCluster(cluster)])

  expect(getSavedClusters(state)).toEqual([])
})

test("setCluster", () => {
  let state = store.dispatchAll([setCluster(cluster)])

  expect(getCurrentCluster(state)).toEqual(cluster)
})

test("setClusterMessage to message", () => {
  let state = store.dispatchAll([setClusterError("no spaces")])

  expect(getClusterMessage(state)).toEqual("no spaces")
})

test("setClusterMessage to empty", () => {
  let state = store.dispatchAll([setClusterError("")])

  expect(getClusterMessage(state)).toEqual("")
})
