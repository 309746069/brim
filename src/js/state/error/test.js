/* @flow */
import AppError from "../../models/AppError"
import error from "./"
import initTestStore from "../../test/initTestStore"

let store
beforeEach(() => {
  store = initTestStore()
})

test("init state", () => {
  let state = store.getState()
  expect(error.getError(state)).toEqual(null)
  expect(error.getVisible(state)).toBe(false)
})

test("set", () => {
  let e = new AppError()
  let state = store.dispatchAll([error.set(e)])

  expect(error.getError(state)).toEqual(e)
  expect(error.getVisible(state)).toBe(true)
})

test("clear", () => {
  let e = new AppError()
  let state = store.dispatchAll([error.set(e), error.clear()])

  expect(error.getError(state)).toEqual(null)
  expect(error.getVisible(state)).toBe(true)
})

test("dismiss", () => {
  let e = new AppError()
  let state = store.dispatchAll([error.set(e), error.dismiss()])

  expect(error.getError(state)).toEqual(e)
  expect(error.getVisible(state)).toBe(false)
})
