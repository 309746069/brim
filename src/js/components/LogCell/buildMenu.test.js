/* @flow */

import buildMenu from "./buildMenu"
import * as mockLogs from "../../test/mockLogs"
import mockSpace from "../../test/mockSpace"

test("conn log with pcap support", () => {
  const log = mockLogs.conn()
  const dispatch = jest.fn()
  const field = log.getField("id.orig_h")
  const space = mockSpace()
  const menu = buildMenu({log, dispatch, field, space})

  expect(menu).toMatchSnapshot()
})

test("dns log", () => {
  const log = mockLogs.dns()
  const dispatch = jest.fn()
  const field = log.getField("query")
  const space = mockSpace()
  const menu = buildMenu({log, dispatch, field, space})

  expect(menu).toMatchSnapshot()
})

test("time field for weird log", () => {
  const log = mockLogs.weird()
  const dispatch = jest.fn()
  const field = log.getField("ts")
  const space = mockSpace()
  const menu = buildMenu({log, dispatch, field, space})

  expect(menu).toMatchSnapshot()
})

test("time field for conn log", () => {
  const log = mockLogs.conn()
  const dispatch = jest.fn()
  const field = log.getField("ts")
  const space = mockSpace()
  const menu = buildMenu({log, dispatch, field, space})

  expect(menu).toMatchSnapshot()
})
