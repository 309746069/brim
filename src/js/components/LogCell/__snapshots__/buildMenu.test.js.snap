// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`conn log with pcap support 1`] = `
Array [
  Object {
    "onClick": [Function],
    "text": "Filter out these values",
    "type": "action",
  },
  Object {
    "onClick": [Function],
    "text": "Only show these values",
    "type": "action",
  },
  Object {
    "onClick": [Function],
    "text": "Count by id.orig_h",
    "type": "action",
  },
  Object {
    "type": "seperator",
  },
  Object {
    "onClick": [Function],
    "text": "Download PCAPS",
    "type": "action",
  },
  Object {
    "onClick": [Function],
    "text": "View log details",
    "type": "action",
  },
]
`;

exports[`dns log 1`] = `
Array [
  Object {
    "onClick": [Function],
    "text": "Filter out these values",
    "type": "action",
  },
  Object {
    "onClick": [Function],
    "text": "Only show these values",
    "type": "action",
  },
  Object {
    "onClick": [Function],
    "text": "Count by query",
    "type": "action",
  },
  Object {
    "type": "seperator",
  },
  Object {
    "onClick": [Function],
    "text": "View log details",
    "type": "action",
  },
]
`;

exports[`time field for conn log 1`] = `
Array [
  Object {
    "onClick": [Function],
    "text": "Use as \\"start\\" time",
    "type": "action",
  },
  Object {
    "onClick": [Function],
    "text": "Use as \\"end\\" time",
    "type": "action",
  },
  Object {
    "type": "seperator",
  },
  Object {
    "onClick": [Function],
    "text": "Download PCAPS",
    "type": "action",
  },
  Object {
    "onClick": [Function],
    "text": "View log details",
    "type": "action",
  },
]
`;

exports[`time field for weird log 1`] = `
Array [
  Object {
    "onClick": [Function],
    "text": "Use as \\"start\\" time",
    "type": "action",
  },
  Object {
    "onClick": [Function],
    "text": "Use as \\"end\\" time",
    "type": "action",
  },
  Object {
    "type": "seperator",
  },
  Object {
    "onClick": [Function],
    "text": "View log details",
    "type": "action",
  },
]
`;
